src/bp/ui-studio/src/web/components/Confusion/index.tsx

```diff
@@ -0,0 +1,159 @@
+}
+
+type Health = 'unknown' | 'bad' | 'medium' | 'good'
+
+interface HealthInfo {
+  f1: number
+  health: Health
+}
+
+interface State {
+  health: Health
+}
+
+export default class NluPerformanceStatus extends React.Component<Props, State> {
+  state: State = {
+    health: 'unknown'
+  }
+  status: NluStatus = {
+    f1score: undefined,
+    synced: true,
+    computing: false
+  }
+
+  componentDidMount() {
+    this.fetchConfusion()
+  }
+
+  componentDidUpdate() {
+    if (!this.props.synced && this.status.synced) {
+      this.status.synced = false
+      this.setState({ health: 'unknown' })
+    }
+  }
+
+  fetchConfusion = (modelHash?) => {
+    this.getConfusionMatrix(modelHash)
+      .then(matrixInfo => {
+        const { matrix, confusionComputing } = matrixInfo
+        if (confusionComputing) {
+          this.setState({ health: 'unknown' })
+
+          this.status = {
+            f1score: undefined,
+            synced: false,
+            computing: true
+          }
+        } else {
+          const { f1, health } = this.extractHealthFromMatrix(matrix)
+          this.setState({ health })
+
+          this.status = {
+            f1score: f1,
+            synced: !!matrix,
+            computing: false
+          }
+        }
+        this.props.updateNluStatus(this.status)
+      })
+      .catch(this.handleError)
+  }
+
+  getConfusionMatrix: (h?: string) => Promise<MatrixInfo> = async (modelHash?) => {
+    if (!modelHash) {
+      const modelHashResponse = await axios.get(`${window.BOT_API_PATH}/mod/nlu/currentModelHash`)
+      modelHash = modelHashResponse.data
+    }
+
+    return axios
+      .get(`${window.BOT_API_PATH}/mod/nlu/confusion/${modelHash}`)
+      .then(confusionMatrixResponse => {
+        return confusionMatrixResponse.data as MatrixInfo
+      })
+      .catch(err => {
+        return { matrix: undefined, confusionComputing: false }
+      })
+  }
+
+  extractHealthFromMatrix(matrix): HealthInfo {
+    if (!matrix || !matrix.intents || !matrix.intents.all) {
+      return { f1: undefined, health: 'unknown' }
+    }
+
+    const f1 = matrix.intents.all.f1 as number
+    if (f1 < BAD_UPPER_BOUND) {
+      return { f1, health: 'bad' }
+    }
+
+    if (f1 >= GOOD_LOWER_BOUND) {
+      return { f1, health: 'good' }
+    }
+
+    return { f1, health: 'medium' }
+  }
+
+  calculateConfusion = async () => {
+    const { confusionComputing } = await this.getConfusionMatrix()
+
+    this.status.computing = this.status.computing || confusionComputing
+    if (!this.status.computing) {
+      this.status.computing = true
+      this.props.updateNluStatus(this.status)
+      this.setState({ health: 'unknown' })
+
+      axios
+        .post(`${window.BOT_API_PATH}/mod/nlu/confusion`)
+        .then(res => {
+          this.status.computing = false
+          this.props.updateNluStatus(this.status)
+          this.fetchConfusion(res.data.modelHash)
+        })
+        .catch(this.handleError)
+    }
```

Comment:
We try to avoid using then() as much as possible, and convert existing ones to async/await when we refactor them. 
```js
try {
  const { data} = await axios.post...
  //etc
} catch(err){
console.error('...')
}